datetime/src/main/java/com/fasterxml/jackson/datatype/jsr310/deser/JavaTimeDeserializerModifier.java

```diff
@@ -0,0 +1,32 @@
+package com.fasterxml.jackson.datatype.jsr310.deser;
+
+import java.time.Month;
+
+import com.fasterxml.jackson.databind.BeanDescription;
+import com.fasterxml.jackson.databind.DeserializationConfig;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.deser.BeanDeserializerModifier;
+
+/**
+ * @since 2.17
+ */
+public class JavaTimeDeserializerModifier extends BeanDeserializerModifier {
+    private final boolean _oneBaseMonths;
+
+    public JavaTimeDeserializerModifier() {
+        this(false);
+    }
+
+    public JavaTimeDeserializerModifier(boolean oneBaseMonths) {
+        _oneBaseMonths = oneBaseMonths;
+    }
+
+    @Override
+    public JsonDeserializer<?> modifyEnumDeserializer(DeserializationConfig config, JavaType type, BeanDescription beanDesc, JsonDeserializer<?> defaultDeserializer) {
+        if (type.hasRawClass(Month.class)) {
+            return new MonthDeserializer((JsonDeserializer<Enum>) defaultDeserializer, _oneBaseMonths);
```

Comment:
Shouldn't this only need to be returned if `_oneBasedMonths` is true? That might simplify handling a bit when there are fewer checks.

Reply:
@cowtowncoder - it's mostly about the naming and meaning of the classes. If we do it like this, perhaps it's better to rename the deserializer from `MonthDeserializer` to `OneBasedMonthDeserializer` and make it always updated the value returned by the delegate (without checking the `_oneBasedMonths ` flag -- and move the check here).
What do you think?

Reply:
Yes, I think that makes sense -- I like explicit naming.